### ISSUE TITLE
Use InputCapture for Mouse receiver plugin

### ISSUE THREAD

**Issue by sidevesh** [GSConnect/gnome-shell-extension-gsconnect#1676]
### Describe your request

Current UX for sharing laptop's mouse and keyboard with paired device vie the Mouse receiver plugin is very cumbersome,
using the newly added in GNOME 45 https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/2628 InputCapture this can be made a lot better since we can just capture alll input and pass it on.
Even something like Mac OS Universal Control would be possible where moving the mouse to a set edge of screen starts capturing and passing on inputs to paired device, making it as if the cursor just moved to the next device.

### GSConnect version

1.28.0

### Installed from

GNOME Extensions website

**Comment by ferdnyc** [comment]
> Even something like Mac OS Universal Control would be possible where moving the mouse to a set edge of screen starts capturing and passing on inputs to paired device, making it as if the cursor just moved to the next device.

Mmmm, _would_ it? Input-capture is purely a one-way (sending) protocol, correct? There's no feedback from the other end, AIUI. (Unless maybe it's used in conjunction with Remote Desktop as the return path.)

That might be able to come _close_, but for a pointer "teleport" feature to really work seamlessly (the way the old Synergy daemon did, cross-platform), you really need two-way communication. The input-sending computer needs to be able to track the pointer on the far end, and be sent updates of its position. Otherwise, there's no way to know when it's hit the far edge of the screen and stopped moving, which makes it impossible to correctly compute the distance you need to travel _back_ before you reenter the original display.

I mean, I'm not saying it's not a worthwhile idea _anyway_, but seamless warping between separate displays requires more than just delivering pointer motions to the target device.

**Comment by sidevesh** [comment]
Makes sense, although I think it still should be possible to implement something very close even with one way data flow,
rather than relying on the cursor moving back through the edge on the paired phone to stop remote input,
we can just use a keyboard shortcut combination,
gsconnect can detect when we move the cursor to one of the preconfigured edge of screen,
at that point, start remote input and show a persistent modal telling the user the keyboard shortcut to use to stop sharing mouse and keyboard,
this modal can also relay  other info like the Remote keyboard not being active for example as subtext,
something like this:
![Screenshot from 2024-03-13 16-54-01](https://github.com/GSConnect/gnome-shell-extension-gsconnect/assets/17254413/a37aa4ba-a5b7-459f-92d8-18ea56670200)
